src/main/resources/com/google/api/codegen/go/sample.snip

```diff
@@ -155,12 +164,42 @@
   }
 @end
 
-@private callChain(methodNameComponents)
-  @join resource : methodNameComponents on "."
+@private field(field)
+  @if field.description
+    @join line : util.getDocLines(field.description, 100)
+      // {@line}
+    @end
+
+
+  @end
+  {@field.name} := {@field.defaultValue}  // TODO: Update placeholder value.
+  @if field.example
+    // {@field.example}
+  @end
+@end
+
+@private callChain(class)
+  {@resourceCallChain(class)}({@paramList(class.methodCallFieldVarNames)}){@optionalFieldCallChain(class)}
+@end
+
+@private resourceCallChain(class)
+  @join resource : class.methodNameComponents on "."
     {@resource}
   @end
 @end
 
+@private optionalFieldCallChain(class)
+  @join field : class.optionalFields on EMPTY
```

Comment:
Okay, maybe let's file an issue for this; it had been on my list since long ago to deal with line breaking more generally, for instance with long argument lists. The joining mechanisms in the snippet engine were designed with this in mind, so it's probably just a question of properly leveraging that machinery.